src/retry/retry.go

```diff
@@ -15,26 +15,44 @@
 package retry
```

Comment:
Been a while since I had to think about implementing backoff for aws apis

Reply:
lol yeah im leaving it for now, but itll likely die